src/plugins/intel_gna/src/transformations/gather_sinking_unary.cpp

```diff
@@ -0,0 +1,205 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/gather_sinking_unary.hpp"
+
+#include <transformations/utils/utils.hpp>
+#include <utility>
+#include <openvino/cc/ngraph/itt.hpp>
+
+#include "openvino/opsets/opset9.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/gather_sinking_utils.hpp"
+#include "transformations/rt_info/gather_sinking_attr.hpp"
+
+using namespace ov;
+using namespace ov::opset9;
+using namespace ov::pass::pattern;
+using namespace ov::op::util;
+using namespace gather_sinking;
+
+namespace {
+
+using NodePtr = std::shared_ptr<ov::Node>;
+using NodePair = std::pair<NodePtr, NodePtr>;
+
+/**
```

Comment:
few methods below loos almost the same as in
https://github.com/openvinotoolkit/openvino/blob/master/src/common/transformations/src/transformations/common_optimizations/transpose_sinking_unary.cpp

I just wonder if there is a chance to reuse the code instead of copying it to gna.

Reply:
There was a discussion about it offline. It is preferable to reuse TransposeSinking code, specially src/common/transformations/src/transformations/common_optimizations/transpose_sinking_utils.cpp.
In such a case we have to add GatherSinking into it. It is not possible just to reuse existed code. But TransposeSinking is placed in common transformations, that is used by entire OpenVINO. There are special requirements to common optimization transformations. We currently place GatherSinking into GNA plugin to solve GNA-plugin tasks. Moving it to the common optimization is possible but it requires a special work on it. It will be done later.

Reply:
We decided not to use any code from the TransposeSinking.